### PR TITLE
chore(sdk): remove retrofit and okhttp

### PR DESCRIPTION
**Pull request by anonvt**
- [x] Added native alternative for Http POST
- [x] Tested with localhost 
- [x] Check response for 200
- [x] Remove Retrofit and Okhttp

~~Test Bearer token Authorization~~
~~Add mocked test~~
Will be in future PRs

Closes: https://topsort.atlassian.net/browse/API-796

### CODE REVIEW

File: TopsortAnalytics/src/main/java/com/topsort/analytics/core/HttpClient.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        val requestedURL: URL = try {
+            URL(url)
+        } catch (e: MalformedURLException) {
+            val error = IOException("Attempted to use malformed url: $url", e)
+            throw error
+        }
+        val connection = requestedURL.openConnection() as HttpURLConnection
+        connection.connectTimeout = 15_000 // 15s
+        connection.readTimeout = 20_1000 // 20s
+        //connection.doInput = true
```
Comment (jbergstroem) [inline TopsortAnalytics/src/main/java/com/topsort/analytics/core/HttpClient.kt:103]: Do we want this or not? If no, remove commented code please

Comment (anonvt) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/core/HttpClient.kt:103]: I'll remove it

File: TopsortAnalytics/src/main/java/com/topsort/analytics/core/HttpClient.kt [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import java.util.zip.GZIPOutputStream
+
+const val LIBRARY_VERSION = 1.0
+
+data class HttpResponse(
+    val code : Int,
+    val message : String,
+){
+    fun isSuccessful() : Boolean {
+        return code in 200..299
```
Comment (jbergstroem) [inline TopsortAnalytics/src/main/java/com/topsort/analytics/core/HttpClient.kt:18]: could do startsWith as well

File: TopsortAnalytics/src/main/java/com/topsort/analytics/service/TopsortAnalyticsHttpService.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+import com.google.gson.Gson
+import com.topsort.analytics.Cache
+import com.topsort.analytics.core.HttpClient
+import com.topsort.analytics.core.HttpResponse
+import com.topsort.analytics.model.ClickEvent
+import com.topsort.analytics.model.ImpressionEvent
+import com.topsort.analytics.model.PurchaseEvent
+
+private const val apiUrl = "https://api.topsort.com/v1/events"
```
Comment (jbergstroem) [inline TopsortAnalytics/src/main/java/com/topsort/analytics/service/TopsortAnalyticsHttpService.kt:11]: Can the hostname be a project wide constant and potentially overridable? Lets say we wanna point it at a staging or test environment

File: TopsortAnalytics/src/test/java/com/topsort/analytics/TopsortAnalyticsHttpServiceTest.kt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+internal class TopsortAnalyticsHttpServiceTest {
+
+    private lateinit var analytics: TopsortAnalytics
+    private lateinit var service: TopsortAnalyticsHttpService
+    private lateinit var session: Session
+
+    @Before
+    fun setup() {
+        analytics = Analytics
+        service = TopsortAnalyticsHttpService
```
Comment (jbergstroem) [inline TopsortAnalytics/src/test/java/com/topsort/analytics/TopsortAnalyticsHttpServiceTest.kt]: Would be great to mock a server inc responses; i don't think we wanna fire to a prod/test environment here.

File: TopsortAnalytics/src/main/java/com/topsort/analytics/core/HttpClient.kt (hunk 2 above)
Comment (anonvt) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/core/HttpClient.kt:18]: `code` returned by the native `HttpURLConnection` is an Int

**Review by PabloReszczynski — APPROVED** [review]

File: TopsortAnalytics/src/main/java/com/topsort/analytics/service/TopsortAnalyticsHttpService.kt (hunk 3 above)
Comment (anonvt) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/service/TopsortAnalyticsHttpService.kt:11]: I'll address in future PRs when we get a better testing setup

Comment (jbergstroem) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/service/TopsortAnalyticsHttpService.kt:11]: This is not about tests though

File: TopsortAnalytics/src/main/java/com/topsort/analytics/core/HttpClient.kt [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    @Throws(IOException::class)
+    override fun close() {
+        connection.disconnect()
+    }
+}
+
+internal fun HttpURLConnection.createPostConnection(): Connection {
+    val encoding = getRequestProperty("Content-Encoding") ?: ""
+    val outputStream: OutputStream =
+        if (encoding.contains("gzip")) {
```
Comment (jbergstroem) [inline TopsortAnalytics/src/main/java/com/topsort/analytics/core/HttpClient.kt:59]: do we additionally wanna support deflate, br here? can the http library help us in any way?

**Pull request merged**